### DISCUSSION TITLE
Local variables highlighting

### DISCUSSION THREAD

**Discussion by susliko** [tlaplus-community/tree-sitter-tlaplus#47] [Q&A]
@ahelwer Hi🖖🏻 
Tree-sitter [has](https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables) a local-variable highlighting feature, but I was unable to discover whether nvim integration with tree-sitter supports it.
Do you know anything about it?

Also, this repo has [locals.scm](https://github.com/tlaplus-community/tree-sitter-tlaplus/blob/main/queries/locals.scm) queries, but they are missing in the main tree-sitter [repo](https://github.com/nvim-treesitter/nvim-treesitter/tree/master/queries/tlaplus). Why so?

**Comment by ahelwer** [comment]
Not yet supported in neovim afaik, see https://github.com/nvim-treesitter/nvim-treesitter/discussions/1667

**Reply by susliko** [reply]
Happy to share some results here.
Defining `locals.scm` as
```scheme
(constant_declaration (identifier) @definition.constant)
(pcal_macro parameter: (identifier) @definition.parameter)
(pcal_var_decl (identifier) @definition.variable)
; etc
```
and `highlights.scm` as
```scheme
((identifier_ref) @constant.builtin (#is? @constant.builtin constant))
((identifier_ref) @parameter (#is? @parameter parameter))
((identifier_ref) @variable.builtin (#is? @variable.builtin variable))
;etc
```
results in properly highlited references:
![image](https://user-images.githubusercontent.com/15569000/148730643-a958b9f2-0330-49a4-a6ce-061aa916ce96.png)

Performance costs are yet to be measured

**Reply by ahelwer** [reply]
Oh wonderful! If you'd like to define the locals.scm file, you can open a PR against the [nvim/queries/tlaplus](https://github.com/tlaplus-community/tree-sitter-tlaplus/tree/main/nvim/queries/tlaplus) directory of this repo (I use it as the source of truth for the neovim queries) and then similarly the [queries/tlaplus](https://github.com/nvim-treesitter/nvim-treesitter/tree/master/queries/tlaplus) directory of the nvim-treesitter repo.
